javascript/javascript1/week3/phone_usage.js

```diff
@@ -0,0 +1,102 @@
+//phone usage
+const activities = [];
+const USER_LIMIT = 150;
+const WARNING_THRESHOLD = USER_LIMIT * 0.8;
+const categories = {
+  Social: ["Instagram", "Tik Tok", "Messenger", "WhatsApp"],
+  Intertainment: ["Netflix", "Games", "YouTube"],
+  Health: ["Fitbit"],
+};
+let totalDuration = 0;
+
+function addActivity(activity, duration) {
+  const date = new Date().toLocaleDateString();
+  const type = assignCategory(activity);
+  const userActivity = {
+    date,
+    activity,
+    duration,
+    type,
+  };
+  activities.push(userActivity);
+  totalDuration += duration;
+  checkWarningThreshold(totalDuration); //it's probably not the best place for a warning but it'll be nice to notify user about threshold somehow
+  return activities;
+}
+function checkWarningThreshold(time) {
+  if (time >= WARNING_THRESHOLD) {
+    console.warn("Warning: You are approaching your usage limit.");
+  }
+}
+function assignCategory(activity) {
+  for (const category in categories) {
+    if (categories[category].includes(activity)) return category;
+  }
+  return "Other";
+}
+
+function calculateCategoriesTimes() {
+  const categoryTimes = {};
+  for (const activity of activities) {
+    if (!categoryTimes[activity.type]) {
+      categoryTimes[activity.type] = 0; //initialize the property of obj.
+    }
+    categoryTimes[activity.type] += activity.duration;
+  }
+  return categoryTimes;
+}
+
+function calculateMaxTime() {
+  let maxTime = 0;
+  let longestActivity = null;
+  for (let activity of activities) {
+    if (activity.duration > maxTime) {
+      maxTime = activity.duration;
+      longestActivity = {
+        title: activity.activity,
+        duration: activity.duration,
+      };
+    }
+  }
+  return longestActivity;
+}
+
+function showStatus(date) {
+  //date must be correct format though. Is it possible/necessary to be flexible here?
+  let totalTime = 0;
+  let amountOfActivities = 0;
```

Comment:
It took me a moment to understand what is going on in this function. Sorry, I was a bit lazy. First I was confused of why you have those variables when you already have `totalDuration` and you can see the length of the `activities` array. Then I saw that the function gets status for a particular date. I think those variables could be named a bit better to reflect the case.
```suggestion
  let selectedDateTotalTime = 0;
  let selectedDateTotalActivities = 0;
```